### PR TITLE
add ios_pmtud network module

### PR DESCRIPTION
**Pull request by mkomon**
##### SUMMARY
Add ios_pmtud module to perform path MTU discovery on Cisco IOS devices.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
ios_pmtud

##### ANSIBLE VERSION
```
ansible 2.5.0 (devel 6e7cfb5add) last updated 2017/09/17 18:42:39 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/etc/ansible/modules', u'/Users/mkomon/ansible-ios-pmtud', u'/Users/mkomon/Documents/github/ansible-junos-stdlib/library']
  ansible python module location = /Users/mkomon/Documents/github/ansible/lib/ansible
  executable location = /Users/mkomon/Documents/github/ansible/bin/ansible
  python version = 2.7.13 (default, Jul 18 2017, 09:17:00) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

##### ADDITIONAL INFORMATION
Finds path MTU to destination IP, with optional VRF and source IP/interface specification. With defaults it finds MTU in range 989-1500 B in exactly 9 steps (always log2(max_range) steps). Tolerates any packet loss lower than 100 pct.

Tested on
- 15.5(3)S3
- 15.4(3)S4

Example invocation:
```
  - name: Check Internet circuit MTU
      ios_pmtud:
        dest: 8.8.8.8
        provider: "{{ cli }}"
        vrf: internet
        max_range: 256
      register: mtu
```

edit: updated Ansible version

### CODE REVIEW

File: lib/ansible/modules/network/ios/ios_pmtud.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            - Source interface name or IP address
+        required: false
+    max_size:
+        description:
+            - Start and max size for path MTU discovery.
+        default: 1500
+        required: false
+    max_range:
+        description:
+            - Max range of path MTU discovery. Must be 2^n.
```
Comment (gundalow) [inline lib/ansible/modules/network/ios/ios_pmtud.py:42]: Would be be worth adding some code to ensure that `max_range` is `2^n`?

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/ios/ios_pmtud.py (hunk 1 above)
Comment (mkomon) [reply lib/ansible/modules/network/ios/ios_pmtud.py:42]: It certainly would, thanks for input!
I've added a check and revert to default value when check fails.

**Comment by mkomon** [comment]
BTW Juniper [has accepted](https://github.com/Juniper/ansible-junos-stdlib/pull/263) a cousin module for PMTUD on Junos devices.

File: lib/ansible/modules/network/ios/ios_pmtud.py (hunk 1 above)
Comment (mkomon) [reply lib/ansible/modules/network/ios/ios_pmtud.py:42]: @gundalow How do you like the fix? Please let me know if it needs more work, thank you.

**Comment by gundalow** [comment]
Hi, we are getting close to the cut off (15th Jan 2018) for new modules for Ansible 2.5. Looks like this is close, though just needs a few minor things before it can be merged.
Please rebase

**Comment by gundalow** [comment]
Rebase needed http://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html

File: lib/ansible/modules/network/ios/ios_pmtud.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+# Run a quicker path MTU discovery in smaller range
+- name: Test failure of the module
+  ios_pmtud:
+    ip: 8.8.8.8
+    vrf: internet
+    max_range: 64
+    provider: "{{ cli }}"
+
+# fail the module
```
Comment (gundalow) [inline lib/ansible/modules/network/ios/ios_pmtud.py:78]: I think this example can be removed.

File: test/integration/targets/ios_pmtud/tasks/cli.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (gundalow) [inline test/integration/targets/ios_pmtud/tasks/cli.yaml:1]: Please update this to match https://github.com/ansible/ansible/pull/33920

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by gundalow** [comment]
@mkomon Cut off for new functionality for 2.5 is 15th Jan.

If you need any assistance with this feel to ask in `#ansible-network` on Freenode IRC

**Comment by justjais** [comment]
@mkomon  Thanks for raising the PR, can u please implement the review comment and update the version to Ansible `2.9` if you want the module to be merged in Ansible.

**Comment by Akasurde** [comment]
Hi @mkomon, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/cisco.ios<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**